src/sentry/auth/system.py

```diff
@@ -79,7 +79,7 @@ def is_expired(self) -> bool:
     @memoize
     def user(self) -> AnonymousUser:
         user = AnonymousUser()
-        user.is_active = True
+        user.is_active = True  # type: ignore[assignment]
```

Comment:
yikes!  this is bad.  we should make a new user type that has this appropriately set rather than mutating AnonymousUser

the type system tries to enforce this is always `False` and so if it's suddenly `True` we're losing internal integrity

not to mention all anonymous users are considered equal and hash to the same thing -- they're meant to be entirely immutable

Reply:
cool will close and fix this if i get a bit of time